modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientTable.java

```diff
@@ -72,9 +72,9 @@ public class ClientTable implements Table {
 
     private final Object partitionAssignmentLock = new Object();
 
-    private CompletableFuture<List<String>> partitionAssignment = null;
+    private volatile CompletableFuture<List<String>> partitionAssignment = null;
 
-    private long partitionAssignmentVersion = -1;
+    private volatile long partitionAssignmentVersion = -1;
```

Comment:
why do you append volatile ? as i can see all usage is already under monitor.

Reply:
`getPartitionAssignment` uses double-checked locking, which requires `volatile`.